matter/src/crypto/crypto_esp_mbedtls.rs

```diff
@@ -85,6 +85,10 @@ impl KeyPair {
 }
 
 impl CryptoKeyPair for KeyPair {
+    fn get_private_key(&self, priv_key: &mut [u8]) -> Result<usize, Error> {
```

Comment:
I initially started with mbedTLS assuming that it was going to be faster for embedded platforms, but looks like it will take some time to converge. 

There was a suggestion to use rustls (https://github.com/rustls/rustls) and ring (https://github.com/briansmith/ring) which might get us there faster. Just FYI...